openwisp_radius/tests/test_api/test_freeradius_api.py

```diff
@@ -681,24 +684,37 @@ def test_accounting_start_200(self):
         self.assertEqual(RadiusAccounting.objects.count(), 0)
         ra = self._create_radius_accounting(**self._acct_initial_data)
         data = self._prep_start_acct_data()
-        response = self.post_json(data)
+        with catch_signal(radius_accounting_success) as handler:
+            response = self.post_json(data)
+        handler.assert_called_once()
         self.assertEqual(response.status_code, 200)
         self.assertIsNone(response.data)
         self.assertEqual(RadiusAccounting.objects.count(), 1)
         ra.refresh_from_db()
         self.assertAcctData(ra, data)
 
+    @mock.patch(
+        'openwisp_radius.receivers.send_login_email.delay', side_effect=OperationalError
+    )
+    @mock.patch('openwisp_radius.receivers.logger')
+    def test_celery_broker_unreachable(self, logger, *args):
+        data = self._prep_start_acct_data()
+        radius_accounting_success.send(sender=AccountingView, accounting_data=data)
+        logger.warn.assert_called_with('Celery broker is unreachable')
+
     def test_accounting_start_radius_token_201(self):
         self._get_org_user()
         self._login_and_obtain_auth_token()
         data = self._prep_start_acct_data()
         data.update(username='tester')
         self.assertEqual(RadiusAccounting.objects.count(), 0)
-        response = self.client.post(
-            self._acct_url,
-            data=json.dumps(data),
-            content_type='application/json',
-        )
+        with catch_signal(radius_accounting_success) as handler:
```

Comment:
We should also assert that the celery task responsible for sending email is called.

Reply:
I tried that by mocking it but it was failing for the sample app because the delay method was not present for the sample app.

Reply:
That is strange. Can you please add that code and push the changes. We'll find out the reason for failing build after that.

Reply:
@pandafy I have added the tests.